src/spetlr/transformers/validfrom_transformer.py

```diff
@@ -0,0 +1,60 @@
+from datetime import datetime, timezone
+from typing import List
+
+import pyspark.sql.functions as f
+from pyspark.sql import DataFrame, Window
+
+from spetlr.etl import Transformer
+from spetlr.utils import DropOldestDuplicates
+
+
+class ValidFromToTransformer(Transformer):
+    """
+    The class introduces SCD2 columns:
+        ValidFrom
+        ValidTo
+        IsCurrent
+    NB: Be aware, if incremental extraction is used, the logic does not work properly
```

Comment:
As I understand it doesn't work at all if you don't do a fullload? - maybe omit "properly"